tests/suites/os/tests/config-json/index.js

```diff
@@ -0,0 +1,395 @@
+        );
+
+        await this.context
+          .get()
+          .worker.executeCommandInHostOS(
+            `tmp=$(mktemp)&&cat /mnt/boot/config.json | jq 'del(.os.network.wifi)' > $tmp&&mv "$tmp" /mnt/boot/config.json`,
+            this.context.get().link
+          );
+      },
+    },
+    {
+      title: "udevRules test",
+      run: async function (test) {
+        const rule = {
+          99: 'ENV{ID_FS_LABEL_ENC}=="resin-boot", SYMLINK+="disk/test"',
+        };
+
+        await this.context
+          .get()
+          .worker.executeCommandInHostOS(
+            `tmp=$(mktemp)&&cat /mnt/boot/config.json | jq '.os.udevRules=${JSON.stringify(
+              rule
+            )}' > $tmp&&mv "$tmp" /mnt/boot/config.json`,
+            this.context.get().link
+          );
+
+        await rebootDevice(this, test);
+
+        test.is(
+          await this.context
+            .get()
+            .worker.executeCommandInHostOS(
+              "readlink -e /dev/disk/test",
+              this.context.get().link
+            ),
+          await this.context
+            .get()
+            .worker.executeCommandInHostOS(
+              "readlink -e /dev/disk/by-label/resin-boot",
+              this.context.get().link
+            ),
+          "Dev link should point to the correct device"
+        );
+
+        await this.context
+          .get()
+          .worker.executeCommandInHostOS(
+            `tmp=$(mktemp)&&cat /mnt/boot/config.json | jq 'del(.os.udevRules)' > $tmp&&mv "$tmp" /mnt/boot/config.json`,
+            this.context.get().link
+          );
+      },
+    },
+    {
+      title: "sshKeys test",
+      run: async function (test) {
+        await test.resolves(
+          this.context
+            .get()
+            .worker.executeCommandInHostOS(
+              "echo true",
+              this.context.get().link
+            ),
+          "Should be able to establish ssh connection to the device"
+        );
+      },
+    },
+    {
+      title: "persistentLogging configuration test",
+      run: async function (test) {
+        const bootCount = parseInt(
+          await this.context
+            .get()
+            .worker.executeCommandInHostOS(
+              "journalctl --list-boots | wc -l",
+              this.context.get().link
+            )
+        );
+
+        test.comment("Attempting first reboot");
+        await rebootDevice(this, test);
+
+        test.comment("Attempting second reboot");
+        await rebootDevice(this, test);
+
+        const testcount = parseInt(
+          await this.context
+            .get()
+            .worker.executeCommandInHostOS(
+              "journalctl --list-boots | wc -l",
+              this.context.get().link
+            )
+        );
+        test.is(
+          testcount > bootCount,
+          true,
+          `Device should show previous boot records, showed ${testcount} boots`
+        );
```

Comment:
Persistent logging is still failing, I don't completely understand the restructuring of config-json tests but we are waiting for verdict on persistent logging here #2143

Reply:
@vipulgupta2048 I found multiple reasons for persistent logging failing. The first was that the "reboot and check"  procedure was unreliable - after the reboot, we were trying to ssh into the DUT, and if this happened before the DUT was back online, it would result in EHOSTUNREACH. I fix this in this PR

The second reason I saw, was that despite rebooting, the DUT was not consistantly recording the reboots. Sometimes it did sometimes it didn't. I moved the persistent logging test to the end of the config.json tests, as by this time, there should have been around 8 or 9 reboots - so by then, we should surely be able to see previous boot history. 

You can see in these logs: https://jenkins.dev.resin.io/job/leviathan-raspberrypi2/509/artifact/client/build/reports/worker-6f7c7cc-raspberry-pi2-RPi_2%20B%20v1.2.log

That 9 reboots are detected by the test

Reply:
@vipulgupta2048 @rcooke-warwick I have a fix on the way for persistent logging - https://github.com/balena-os/meta-balena/pull/2149

Reply:
@markcorbinuk I see in that PR that the RPI1 test annoyingly failed because of the EHOSTUNREACH error I mentioned. This PR contains a fix that SHOULD prevent that from happening. Between these 2 PRs the persistent logging test should be consistant then